openedx/core/djangoapps/persistenttask/models.py

```diff
@@ -0,0 +1,31 @@
+"""
+Models to support persistent tasks.
+"""
+
+from django.db import models
+
+from openedx.core.lib.properties import JSONProperty
+
+
+class FailedTask(models.Model):
+    """
+    Representation of tasks that have failed
+    """
+    task_name = models.CharField(max_length=255, db_index=True)
+    task_id = models.CharField(max_length=255)
+    argstring = models.TextField(blank=True)
+    kwargstring = models.TextField(blank=True)
+    exc = models.CharField(max_length=255)
+    datetime_failed = models.DateTimeField()
+    datetime_resolved = models.DateTimeField(blank=True, null=True, default=None, db_index=True)
+
+    args = JSONProperty('argstring', allowed_types={list, tuple})
```

Comment:
How are "Properties" different from "Fields"?  

Typically, one expects to see django model `Fields` within a django model.  Consider defining this as a reusable `JSONField` instead, inheriting from django's `Field` class.  This way, you don't need to have separate `argstring` and `args` members on the model.  You can just have an `args` `JSONField` which automatically converts from the underlying datatype (in SQL) to the JSON datatype (in Python).  See how `OpaqueKeyField` or even `DateTimeField` classes are defined.